### PR TITLE
build(update-readme): update README.md packages list dynamically

### PR DESCRIPTION
**Pull request by cbourgois**
### Description of the Change

* `yarn run docs:update-readme` updates availables modules and apps in
`README.md`

104c90ec — build(update-readme): update README.md packages list dynamically

Add `update-readme` script to update [`Available modules`](https://github.com/ovh-ux/manager/blob/master/README.md#available-modules) and [`Available apps`](https://github.com/ovh-ux/manager/blob/master/README.md#available-apps) section in `README.md`.

Examples : 
https://github.com/ovh-ux/manager/blob/build/readme/README.md#availables-modules
https://github.com/ovh-ux/manager/blob/build/readme/README.md#availables-apps

It uses [`markdown-magic`](https://github.com/DavidWells/markdown-magic) to tranform markdown comment, and `lerna list` to retrieve list of packages.

/cc @jleveugle @frenautvh @antleblanc

### CODE REVIEW

File: scripts/readme.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+          headers.push('');
+
+
+          const lines = packageList.map(({ name, location }) => {
+            const line = [
+              `[${name}](https://github.com/ovh-ux/manager/tree/master/${location})`,
+            ];
+
+            if (options.showBadges) {
+              line.push(`[![npm version](https://badgen.net/npm/v/${name})](https://www.npmjs.com/package/${name})`);
```
Comment (antleblanc) [inline scripts/readme.js]: You can use multi-line strings inside template literals.

File: scripts/readme.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+          }
+          const separators = cols.map(col => '-'.repeat(col.length));
+          cols.push('Changelog');
+          separators.push(':-------:');
+
+          headers.push(`| ${cols.join(' | ')} |`);
+          headers.push(`| ${separators.join(' | ')} |`);
+          headers.push('');
+
+
```
Comment (antleblanc) [inline scripts/readme.js]: Extra empty line.

File: package.json [hunk 3]
```diff
@@ -12,6 +12,7 @@
     "build": "lerna run build --stream",
     "clean": "yarn run clean:dist && npx rimraf **/node_modules",
     "clean:dist": "npx rimraf dist packages/**/dist",
+    "docs:readme": "scripts/readme.js",
```
Comment (antleblanc) [inline package.json]: You can also rename your file `readme` and call it like:

```suggestion
    "docs:readme": "scripts/readme",
```

In counterpart `readme` sounds too generic as filename in my opinion. At least I suggest `update-readme`. Does it make sense for you?

Comment (cbourgois) [reply package.json]: yep 👍

File: scripts/readme.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+          };
+
+          const packageList = filterPackages(packages, options);
+
+          if (packageList.length === 0) {
+            return '';
+          }
+
+          const headers = [
+            `# Availables ${options.name}`,
```
[the remark this answers is not in the document]
Comment (cbourgois) [reply scripts/readme.js]: Of course.

File: scripts/readme.js (hunk 1 above)
Comment (cbourgois) [reply scripts/readme.js]: Yes, I push too early.
I want to set `showBadges=version|deps|devDeps|peerDeps` to be more agile : 
If you set `showBadges=version` you will have only `version` badge.

**Pull request merged**